src/UI/COSetupInfo.java

```diff
@@ -0,0 +1,136 @@
+
+  final CommanderInfo[] COTypes;
+  final OptionSelector currentCO;
+
+  final Color[] colors;
+  final OptionSelector currentColor;
+
+  final String[] factions;
+  final OptionSelector currentFaction;
+
+  final OptionSelector currentTeam;
+
+  final AIMaker[] AIs;
+  final OptionSelector currentAI;
+
+  public COSetupInfo(int numPlayers, int thisPlayer, ArrayList<CommanderInfo> COTypeList, Color[] colorList, String[] factionList, ArrayList<AIMaker> AIList)
+  {
+    super(OptionList.values().length);
+    
+    COTypes = COTypeList.toArray(new CommanderInfo[0]);
+    currentCO = new OptionSelector(COTypes.length);
+    currentCO.setSelectedOption(0);
+    // Copied TODO: Consider changing this to sequential or random COs once we have enough.
+    // It might be better to save the last state in which a game was started, and use that.
+
+    colors = colorList;
+    currentColor = new OptionSelector(colors.length);
+    currentColor.setSelectedOption(thisPlayer);
+
+    factions = factionList;
+    currentFaction = new OptionSelector(factions.length);
+    currentFaction.setSelectedOption(thisPlayer);
+
+    currentTeam = new OptionSelector(numPlayers);
+    currentTeam.setSelectedOption(0); // 0 should be interpreted as "no team"
+
+    AIs = AIList.toArray(new AIMaker[0]);
+    currentAI = new OptionSelector(AIs.length);
+    currentAI.setSelectedOption(0); // default to human
+  }
+
+  /**
+   * Keeps the selected option within the normal bounds, and tells you if you're hitting the end of the option set.
+   * @return whether you've hit the option "barrier"
+   */
+  public boolean pickOption(InputAction action)
+  {
+    int previous = this.getSelectionNormalized();
+    this.handleInput(action);
+    
+    // If our absolute and normalized selections are no longer consistent, clamp and notify.
+    if( this.getSelectionAbsolute() != this.getSelectionNormalized() )
+    {
+      this.setSelectedOption(previous);
+      return true;
+    }
+    
+    return false;
+  }
+
+  public OptionSelector getCurrentOptionSelector()
+  {
+    switch (OptionList.values()[this.getSelectionNormalized()])
+    {
+      case COMMANDER:
+        return currentCO;
+      case COLOR:
+        return currentColor;
+      case FACTION:
+        return currentFaction;
+      case TEAM:
+        return currentTeam;
+      case AI:
+        return currentAI;
+      default:
+        return currentCO; // just pretend there's nothing weird going on, and hope nobody notices
+    }
+  }
+  
+  public CommanderInfo getCurrentCO()
+  {
+    return COTypes[currentCO.getSelectionNormalized()];
+  }
+  
+  public Color getCurrentColor()
+  {
+    return colors[currentColor.getSelectionNormalized()];
+  }
+  
+  public String getCurrentFaction()
+  {
+    return factions[currentFaction.getSelectionNormalized()];
+  }
+  
+  public int getCurrentTeam()
+  {
+    return currentTeam.getSelectionNormalized() - 1; // -1 means "no team"
```

Comment:
This means that the actual team number will not match the team number the user chooses. They don't see it again, so it's probably fine, but still odd. It also appears to conflict with the comment back on line 53.

Reply:
A *selection* of 0 is "no team"
A *result* of -1 is "no team"
They come from the same selection value, but are just stated differently.

I agree it's a little odd. In fairness, it should maybe instead be special cased in the CO-generation logic as 0 -> -1, but I dunno.

Reply:
Is there any reason not to use 0 as "no team"? That would avoid the extra math and confusion. I'd suggest defining `NO_TEAM` as a constant somewhere, but that would conflict with using an `OptionSelector` index for the value.
I can also see the case for using `-1` as the sentinel value though, since it is clearly out of bounds. I guess I'm not gonna fight too hard either way on this one.

Reply:
> Is there any reason not to use 0 as "no team"? That would avoid the extra math and confusion. I'd suggest defining `NO_TEAM` as a constant somewhere, but that would conflict with using an `OptionSelector` index for the value.
> I can also see the case for using `-1` as the sentinel value though, since it is clearly out of bounds. I guess I'm not gonna fight too hard either way on this one.

Indexing from zero is natural (err, whole? heh), and I didn't wanna mess with the existing logic.

I think this is where I pull out your "if it ain't broke, don't fix it" argument.